Modules/Logger/src/selxLoggerImpl.cxx

```diff
@@ -18,96 +18,155 @@
  *=========================================================================*/
 
 #include "selxLoggerImpl.h"
-
-#include "boost/log/sources/record_ostream.hpp"
-#include "boost/log/utility/setup/file.hpp"
-#include "boost/log/utility/setup/console.hpp"
-#include "boost/log/utility/setup/common_attributes.hpp"
+#include "spdlog/details/registry.h"
 
 namespace selx
 {
+
 LoggerImpl
-::LoggerImpl()
+::LoggerImpl() : m_Loggers(), m_AsyncQueueSize( 262144 ), m_AsyncQueueOverflowPolicy( spdlog::async_overflow_policy::block_retry ), m_StreamToString()
 {
-  this->m_Logger = boost::log::sources::severity_channel_logger< SeverityType, Logger::ChannelType >();
+  this->SetSyncMode();
+  this->SetPattern( "[%Y-%m-%d %H:%M:%S.%f] [thread %t] [%l] %v" );
+}
 
-  // Add LineID, TimeStamp, ProcessID and ThreadID
-  boost::log::add_common_attributes();
+LoggerImpl
+::~LoggerImpl()
+{
+  this->RemoveAllStreams();
 }
 
+spdlog::level::level_enum
+LoggerImpl
+::ToSpdLogLevel( const LogLevel& level ) {
+  switch (level) {
+    case LogLevel::TRC:
+      return spdlog::level::level_enum::trace;
+    case LogLevel::DBG:
+      return spdlog::level::level_enum::debug;
+    case LogLevel::INF:
+      return spdlog::level::level_enum::info;
+    case LogLevel::WRN:
+      return spdlog::level::level_enum::warn;
+    case LogLevel::ERR:
+      return spdlog::level::level_enum::err;
+    case LogLevel::CRT:
+      return spdlog::level::level_enum::critical;
+    case LogLevel::OFF:
+      return spdlog::level::level_enum::off;
+    default:
+      itkGenericExceptionMacro( "Invalid log level." );
+  }
+}
 
+void
 LoggerImpl
-::~LoggerImpl()
+::SetLogLevel( const LogLevel& level ) {
```

Comment:
SetLogLevel should have a stream identifier argument too, for which the log level was set. The log level for std::cerr should typically be different that std::cout.

Reply:
Setting log-level per logger interferes with spdloggers global log level. This must be postponed.

Reply:
postponed to #112